packages/fetchai/connections/soef/connection.py

```diff
@@ -475,6 +482,7 @@ async def disconnect(self) -> None:
             self.connection_status.is_connected or self.connection_status.is_connecting
         ), "Call connect before disconnect."
         assert self.in_queue is not None
+        self.channel.disconnect()
```

Comment:
The channel has no concept of disconnect, it is a http client only

Reply:
I can move the code in the un-register function that has the `NotImplementedError` but the `TODO` there says to add the ping functionality there.

Reply:
What you can do is implement a `disconnect` method in the channel which looks like your current unregister, in that it unregisters all services of the agent.